src/models/state/mod.rs

```diff
@@ -1472,6 +1482,92 @@ impl GlobalState {
         // Ok(())
     }
 
+    /// Returns (parent, child) pair of blocks.
+    pub(crate) async fn fetch_block_pair(&self, child_digest: Digest) -> Option<(Block, Block)> {
+        let child = self
+            .chain
+            .archival_state()
+            .get_block(child_digest)
+            .await
+            .expect("fetching block from archival state should work.");
+        let Some(child) = child else {
+            warn!("Got sync challenge for unknown tip");
```

Comment:
This log message is not appropriate now that this function has been moved to `GlobalState`.